tests/test_ast_symbol_extractor.py

```diff
@@ -128,31 +153,29 @@ def xyz(self):
             "symbols_in_volume": {"numpy.twos"},
             "type": "function",
         },
+        "mm.np": {"shadows": "numpy", "lineno": None, "type": "import"}
     }
 
 
-@pytest.mark.xfail()
 def test_import_adds_symbols():
-    # np should be a symbol in the surface area since it could be
-    # imported from this code issue #23
     code = """
     import numpy as np
     from abc import xyz as l
-    from abc import xyz
+    from ggg import efg
+    import ghi
 
     z = np.ones(5)
     """
     tree = ast.parse(dedent(code))
     z = SymbolFinder(module_name="mm")
     z.visit(tree)
-    # TODO: should we add a key in the metadata to say that a symbol is
-    #  a reference to another symbol?
     assert z.symbols == {
-        "mm.np": {},
-        "mm.xyz": {},
-        "mm.l": {},
-        'mm': {'lineno': None, 'symbols_in_volume': set(), 'type': 'module'},
-        "mm.z": {"lineno": 4, "symbols_in_volume": {"numpy.ones"}, "type": "constant"},
+        "mm.np": {"shadows": "numpy", "lineno": None, "type": "import"},
```

Comment:
@CJ-Wright would like your comments on this data model.
Do we want all values of symbols to have the same attributes?
The way it currently stands the `type` determines the attributes in the object, we might consider lifting it out of the values and into a first-class citizen.

Something like this keeps it easy to iterate over all while also simple to filter out unneeded types, would also help keeping all the types data model consistent with each other.
Also may help expose/consume later on via automated tools/schemas. 😄 
```python
z.symbols == {
        "mm.np": {
                "type": "import",
                "data": {"shadows": "numpy", "lineno": None,},
        },
        "mm.f": {
                "type": "function",
                "data": {"symbols_in_volume": set(), "lineno": 5}
       }
```

Reply:
That looks great! It most likely also helps with less flexable json consumers that expect the keys to be consistent in the data model (or at least consistent to a particular type in this case).